Symco 20 student landing page

devin-w-hill opened the pull request:
This is the one. Took out the migrations and a little .png that no one needs. This is the Student Landing Page using django_tables2 and Katie's nav bar. Looks like the models haven't been added to main yet so this includes those as well.

Commit:
Added models for Users app. In progress...

kaigoos committed:
Added new conversation template models

kaigoos committed:
Mob Session: Just changed Assignment class to inherit models.Model

kaigoos committed:
Mob Session: Just changed Assignment class to inherit models.Model

kaigoos committed:
Merge branch 'models' of https://github.com/Likhovodov/Simulated-Conversations into models

kaigoos committed:
Removed views.py and added a view folder

Commit:
Cleaned up models in both apps. Removed references in model files to other model files in the same app. Fixed typos.

Commit:
Cleaned up models in both apps. Removed references in model files to other model files in the same app. Fixed typos.

Commit:
test

Commit:
test

Commit:
set changes to pick up on other comp

devin-w-hill committed:
cleaned up most models. error in student.py on foreignKey relationship with researcher.py. do not attempt to make migrations

devin-w-hill committed:
cleaned up most models. error in student.py on foreignKey relationship with researcher.py. do not attempt to make migrations

devin-w-hill committed:
cleaned up most models. error in student.py on foreignKey relationship with researcher.py. do not attempt to make migrations

devin-w-hill committed:
changed Student.researcher field to added_by.

devin-w-hill committed:
gave foreignkeys a default of 0. Could not makemigrations without settings defaults and they needed to integers.

devin-w-hill committed:
gave foreignkeys a default of 0. Could not makemigrations without settings defaults and they needed to integers.

devin-w-hill committed:
gave all UUIDfields a default of uuid.uuid4. had to import uuid for all models with UUIDfield

devin-w-hill committed:
Changed related name for destination_node

from destination_choices to parent_choices

devin-w-hill committed:
added foreignkey to TemplateResponse

devin-w-hill committed:
updated on_delete behavior

from DO_NOTHING to CASCADE

devin-w-hill committed:
updated field names

from templates to conversation_templates, from labels to subject_labels

devin-w-hill committed:
update field name

from file_name to label_name

devin-w-hill committed:
changed relationship to Researcher

Changed from ManyToManyField to ForeignKey because labels will not be shared between researchers.

devin-w-hill committed:
removed TODO

accessor clash has been resolved so comment stating error was removed.

devin-w-hill committed:
Added docstring 

Simple docstring describes the class and its fields.

Commit:
registered models in admin.py in nested simcon_project.

Commit:
registered models in admin.py in nested simcon_project.

Commit:
Merge remote-tracking branch 'origin/SYMCO-20-student-landing-page' into SYMCO-20-student-landing-page

Commit:
Created a very simple student landing page that displays only the templates that have been assigned to them.

Commit:
Created a very simple student landing page that displays only the templates that have been assigned to them.

devin-w-hill committed:
gave all UUIDfields a default of uuid.uuid4. had to import uuid for all models with UUIDfield

devin-w-hill committed:
gave all UUIDfields a default of uuid.uuid4. had to import uuid for all models with UUIDfield

devin-w-hill committed:
made the templates shown to user paginated.

devin-w-hill committed:
made the templates shown to user paginated.

devin-w-hill committed:
Added a sidebar (with settings gear), footer, and gave the template table its own div. The page now appears similar to the wireframe.

devin-w-hill committed:
Added a sidebar (with settings gear), footer, and gave the template table its own div. The page now appears similar to the wireframe.

devin-w-hill committed:
Committed users static directory

devin-w-hill committed:
Committed users static directory

Commit:
Added settings view and link to it from gear icon

Commit:
Added settings view and link to it from gear icon

Commit:
Made each template name link to a corresponding template start. Template start not fully built out yet.

Commit:
Made each template name link to a corresponding template start. Template start not fully built out yet.

Commit:
Updated table to use django_tables2. Same functionality but now with sortable columns. Added nav bar from Katie's branch and removed the sidebar.

Commit:
Updated table to use django_tables2. Same functionality but now with sortable columns. Added nav bar from Katie's branch and removed the sidebar.

Commit:
changed required python version in pipfile

Commit:
changed gitignore

Commit:
changed gitignore

devin-w-hill committed:
don't need migrations

devin-w-hill committed:
don't need migrations

devin-w-hill committed:
don't need migrations

devin-w-hill committed:
don't need migrations

devin-w-hill committed:
don't need migrations

devin-w-hill committed:
don't need migrations

devin-w-hill committed:
don't need migrations

devin-w-hill committed:
don't need migrations

devin-w-hill committed:
don't need migrations

devin-w-hill committed:
don't need migrations

devin-w-hill committed:
don't need

devin-w-hill committed:
don't need

kaigoos commented:
If we are going to have a base.html in the future I think it's okay to not add the navbar for now since it will be part of the base html file and it's gonna have to be taken out from the things you worked on anyways.

devin-w-hill commented:
@kaigoos that's true...easy enough to take out

devin-w-hill committed:
removed nav bar

nav bar will be included in base.html

devin-w-hill committed:
removed nav bar

nav bar will be included in base.html

kaigoos commented:
Also it says all these files are being added for the first time, but I'm pretty sure some of it is already in main so I'm not 100% sure what is going on. Do you happen to know the reason why?

rosasck commented on .gitignore:
```diff
@@ -5,6 +5,7 @@
 __pycache__
 db.sqlite3
 media
+migrations
```
did you pull from main after we made the new migrations?

devin-w-hill replied:
I must not have.

rosasck commented on Pipfile:
```diff
@@ -0,0 +1,11 @@
+[[source]]
```
make sure to remove this or add to the git ignore! keep in mind we are not keeping this up to date

rosasck commented on simcon_project/conversation_templates/models/__init__.py:
```diff
@@ -0,0 +1,6 @@
+from .conv_template import ConversationTemplate 
```
hmmm i feel like ur git flow might have deviated from master.... im afraid of what will get overriden. if you could not push files you didnt write inside. so remove them from ur git add

rosasck commented on simcon_project/conversation_templates/models/conv_template.py:
```diff
@@ -0,0 +1,24 @@
+from django.db import models
```
same comment as above

rosasck commented on simcon_project/conversation_templates/models/template_node.py:
```diff
@@ -0,0 +1,11 @@
+from django.db import models
```
not sure if this file has a new line at the end...

rosasck commented on simcon_project/conversation_templates/models/template_node_choice.py:
```diff
@@ -0,0 +1,9 @@
+from django.db import models
```
same comment

rosasck commented on simcon_project/simcon_project/forms.py:
```diff
@@ -0,0 +1,16 @@
+from django.contrib.auth.forms import UserCreationForm, UserChangeForm
```
new line at the end of file plzz

rosasck commented on simcon_project/templates/settings_view.html:
```diff
@@ -0,0 +1,35 @@
+<!DOCTYPE html>
```
new line at the end of this file plz

rosasck commented on simcon_project/templates/settings_view.html:
```diff
@@ -0,0 +1,35 @@
+    <meta name="description" content="Created for Simulated Conversations Senior Capstone Project">
+
+    <!-- prevent line breaks when using form fields -->
+    <style type="text/css">
```
i wonder if we wanna keep all of our style in a separate file? i know i had to have a static routed css

devin-w-hill replied:
I've got the real css for my home page stored in its own file in a directory for static assets.

rosasck commented on simcon_project/templates/settings_view.html:
```diff
@@ -0,0 +1,35 @@
+
+    <!-- Placed at the end of the document so the pages load faster -->
+  </body>
+</html>
```
new line at the end

rosasck commented on simcon_project/templates/student_view.html:
```diff
@@ -1,17 +1,16 @@
     <meta charset="utf-8">
     <meta http-equiv="X-UA-Compatible" content="IE=edge">
-    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
```
hmm... not sure if i like the shrunk to fit to be set to no.. that  could scale very poorly

devin-w-hill replied:
It's probably better to have it switched on, yeah.

rosasck commented on simcon_project/templates/student_view.html:
```diff
@@ -1,17 +1,16 @@
-    </style>
-    
+    <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.4.1/css/bootstrap.min.css">
+    <link rel="stylesheet" href="{% static 'users/student_home.css' %}">
```
you also put a static css, put all style in here! Allows very simple organization if we ever wanna update we dont have to search 16 files

devin-w-hill replied:
Combining those two into one would make a mess of the static file. It's not too difficult to figure out what is affecting a particular element using your browser's inspect feature and multiple files allows for better organization I think. Otherwise everything ends up in one big pile.

rosasck commented on simcon_project/templates/student_view.html:
```diff
@@ -21,15 +20,14 @@
+    <div class="body-container">
+          {% render_table table %}
+        <footer>
+          <p>Copyright 2020 Portland State University</p>
```
not sure if we actually have copy right due to our license @Likhovodov

devin-w-hill replied:
True, not sure about this. I added it in to make it look more official early on but probably will need to be changed at some point. I think it's ok for now?

rosasck replied:
yeah!

rosasck commented on simcon_project/templates/template_start_view.html:
```diff
@@ -0,0 +1,35 @@
+
+    <!-- Placed at the end of the document so the pages load faster -->
+  </body>
+</html>
```
new line

rosasck commented on simcon_project/templates/template_start_view.html:
```diff
@@ -0,0 +1,35 @@
+<!DOCTYPE html>
```
you may need to merge or un add these from ur PR... if you push after anybody u will overirde their files

devin-w-hill replied:
Not sure I totally understand what you're getting at. Can't I just merge in the actual page whenever it gets written and then there'd be no danger of me overwriting anything with this placeholder? Removing this would break a piece of my page.

rosasck replied:
thought this was not one of ur pages.. thsi was a response... not sure what happened but it was me saying i understand i thought originally this wasnt one of ur pages but makes sense now!

rosasck commented on simcon_project/users/admin.py:
```diff
@@ -28,4 +29,11 @@ class CustomUserAdmin(UserAdmin):
+    try:
+        admin.site.register(model)
+    except admin.sites.AlreadyRegistered:
+        pass
```
im confused with what to PR... anyway you could leave a comment what files u altered?

rosasck commented on simcon_project/users/models/custom_user.py:
```diff
@@ -53,13 +54,15 @@ def create_superuser(self, email, password, **extra_fields):
     email = models.EmailField(_('email address'), unique=True)
-    first_name = models.CharField(max_length=30)
-    last_name = models.CharField(max_length=30)
+    first_name = models.CharField(max_length=30, default="None")
```
good catch!

rosasck commented on simcon_project/users/static/users/student_home.css:
```diff
@@ -0,0 +1,29 @@
+.body-container {
```
Please comment at each block what component its referring to and what page

rosasck commented on simcon_project/users/tables/__init__.py:
```diff
@@ -0,0 +1 @@
+from .student_home_table import StudentHomeTable
```
hmmm, i dont think u needed to do this. if you just queried for all objects in ur view, and passed the object to the html page, you can render super easily and no need for extra files

devin-w-hill replied:
I tried for a good few hours to pull the information I wanted without using a separate file. I agree that it's probably possible but the documentation for django_tables2 is a little spotty so I couldn't figure it out. The examples they have don't cover the case of pulling data from multiple models using the relationships between them. If you know how to do that, I'd be interested in seeing how

rosasck commented on simcon_project/users/tables/student_home_table.py:
```diff
@@ -0,0 +1,10 @@
+import django_tables2 as tables
```
same comment as above. If we decide to keep this please add a comment in the migration that this file will need to be updated if the migration is updates as well ... by migration i mean model

devin-w-hill replied:
I would need to add a comment to three different models explaining that it's contents are used in just this one table. Not saying it's a bad idea, but do we want to set that precedent for every table that we create?

rosasck commented on simcon_project/users/views/redirect_from_login.py:
```diff
@@ -1,11 +1,17 @@
+from django.urls import reverse
+from django.contrib.auth.decorators import login_required
 
+@login_required
```
does this work? if it does please let @JerCarnahan so he can uncomment these out

devin-w-hill replied:
It worked but I didn't make a real effort to try and break it. I think Jeremy knows, it's been addressed in his PR.

rosasck commented on simcon_project/users/views/researcher_home.py:
```diff
@@ -1,4 +1,7 @@
+
+@login_required
 def ResearcherView(request):
     return render(request, 'researcher_view.html')
```
extra new line at the end of this file

rosasck commented on simcon_project/users/views/settings.py:
```diff
@@ -0,0 +1,7 @@
+
+@login_required
+def SettingsView(request):
+    return render(request, 'settings_view.html')
```
extra new line at the end of this file

rosasck commented on simcon_project/users/views/student_home.py:
```diff
@@ -1,4 +1,35 @@
 
+
+@login_required
+def StudentView(request):
```
why filter it? are some columns not wanted to be shown???

devin-w-hill replied:
The filter makes it so the students only see the templates assigned to them, not just all assigned templates in general. Then the values are the specific columns I want to display. I wanted to figure out a way to do all this without creating the StudentHomeTable class but I couldn't get it to work that way. The information was getting retrieved correctly but not getting fed into the html.

rosasck commented on simcon_project/users/views/student_home.py:
```diff
@@ -1,4 +1,35 @@
+            'conversation_templates__template_responses__completion_date',
+            'conversation_templates__id')
+    template_table = StudentHomeTable(contents)
+    template_table.paginate(page=request.GET.get("page", 1), per_page=10)
```
do we wanna only paginate 10? i did 20 but this could be something we make even throughout! @Likhovodov what should out pagination # be?

devin-w-hill replied:
Yeah I just picked a number, but we should agree on a set value probably.

rosasck commented on simcon_project/users/views/student_home.py:
```diff
@@ -1,4 +1,35 @@
+    template_table = StudentHomeTable(contents)
+    template_table.paginate(page=request.GET.get("page", 1), per_page=10)
+    # must pass context as a dictionary
+    return render(request, 'student_view.html', {'table': template_table})
```
if you dont filter it you could just pass the object contents to this!

devin-w-hill replied:
But then I would have the wrong info :(

rosasck commented on simcon_project/users/views/student_home.py:
```diff
@@ -1,4 +1,35 @@
+            'date_assigned',
+            'conversation_templates__template_responses__completion_date',
+            'conversation_templates__id')\
+        .order_by('conversation_templates__template_responses__completion_date', 'date_assigned')
```
dont think we need this order by since we will use the sort, but if we do we should prioritize completion date not date assigned! @Likhovodov what do u think?

rosasck commented on simcon_project/users/views/student_home.py:
```diff
@@ -1,4 +1,35 @@
+    template_page = assigned_templates_paginated.get_page(page_number)
+    # must pass context as a dictionary
+    return render(request, 'student_view.html', {'template_page': template_page})
+'''
```
new line plz

rosasck reviewed (changes requested):
please update some things and then re request by clicking on the little refresh wheel next to my name under reviewers!

Likhovodov reviewed (changes requested):
Ok before we get moving any further we need to make sure that the files in this PR are only files that you worked on. Maybe there was some kind of copy pasting going on or something but files are labeled as changed that shouldn't be. Any clue how that might have happened?... One suspicion that I'm having is that you might have merged the models branch into your branch and then the models branch was squash merged into main? If so idk what to do... we'll need to do hw

devin-w-hill committed:
pipfile not needed.

devin-w-hill committed:
pipfile not needed.

devin-w-hill committed:
not a changed file.

devin-w-hill committed:
not a changed file.

devin-w-hill committed:
file not changed from master.

devin-w-hill committed:
file not changed from master.

devin-w-hill committed:
added newline to end of file

devin-w-hill committed:
added newline to end of file

devin-w-hill committed:
adding new line to end of file

devin-w-hill committed:
adding new line to end of file

devin-w-hill committed:
no need to merge

devin-w-hill committed:
no need to merge.

devin-w-hill committed:
no need to merge.

devin-w-hill committed:
no need to merge.

devin-w-hill committed:
no need to merge.

devin-w-hill committed:
no need to merge.

devin-w-hill committed:
no need to merge.

devin-w-hill committed:
added newline to end of file

devin-w-hill committed:
added newline to end of file

devin-w-hill committed:
added newline to end of file

devin-w-hill committed:
added newline to end of file

devin-w-hill committed:
changed shrink-to-fit to yes.

Safari browsers will automatically resize page to fit the screen.

devin-w-hill committed:
changed shrink-to-fit to yes.

Safari browsers will automatically resize page to fit the screen.

devin-w-hill committed:
added newline to end of file

devin-w-hill committed:
added newline to end of file

devin-w-hill committed:
models are registered in users/admin.py, not here.

devin-w-hill committed:
models are registered in users/admin.py, not here.

devin-w-hill committed:
not needed

devin-w-hill committed:
not needed

devin-w-hill committed:
not needed

devin-w-hill committed:
not needed

devin-w-hill committed:
not needed

devin-w-hill committed:
not needed

devin-w-hill committed:
not needed

devin-w-hill committed:
not needed

devin-w-hill committed:
not needed.

devin-w-hill committed:
not needed.

devin-w-hill committed:
not needed.

devin-w-hill committed:
not needed.

devin-w-hill committed:
Commented descriptions of what each css rule does

devin-w-hill committed:
Commented descriptions of what each css rule does

devin-w-hill committed:
not needed.

devin-w-hill committed:
not needed.

devin-w-hill committed:
added new line to end of file

devin-w-hill committed:
added new line to end of file

devin-w-hill committed:
new line at end of file

devin-w-hill committed:
new line at end of file

devin-w-hill committed:
removed commented out portion

devin-w-hill committed:
removed commented out portion

devin-w-hill committed:
not needed

devin-w-hill committed:
not needed

devin-w-hill committed:
changed gitignore

devin-w-hill committed:
changed gitignore

devin-w-hill committed:
# Conflicts:
#	simcon_project/conversation_templates/models/__init__.py
#	simcon_project/simcon_project/admin.py
#	simcon_project/simcon_project/settings.py
#	simcon_project/users/admin.py
#	simcon_project/users/models/__init__.py
#	simcon_project/users/models/student.py
#	simcon_project/users/models/subject_label.py

devin-w-hill committed:
# Conflicts:
#	simcon_project/conversation_templates/models/__init__.py
#	simcon_project/simcon_project/admin.py
#	simcon_project/simcon_project/settings.py
#	simcon_project/users/admin.py
#	simcon_project/users/models/__init__.py
#	simcon_project/users/models/student.py
#	simcon_project/users/models/subject_label.py

devin-w-hill committed:
Merge branch 'main' into SYMCO-20-student-landing-page

devin-w-hill commented on simcon_project/simcon_project/admin.py:
```diff
@@ -1,10 +1,3 @@
-admin.site.register(TemplateNodeChoice)
-admin.site.register(TemplateResponse)
-admin.site.register(TemplateNodeResponse)
 
```
models are registered in the admin.py under users.

devin-w-hill committed:
not needed.

devin-w-hill committed:
not needed.

devin-w-hill commented:
Ok so after a making a big mess, I think I've cleaned everything up to where it can be merged. I deleted anything that I didn't make changes to but some of the deleted ones are still hanging around in the Files Changed tab. Not sure how to get rid of them but I don't think they will cause a problem?
Files I actually made changes to/created (total of 16 even though 22 are listed in PR):
.gitignore
simcon_project/simcon_project/admin.py
simcon_project/simcon_project/urls.py
simcon_project/templates/settings_view.html
simcon_project/templates/student_view.html
simcon_project/templates/template_start_view.html
simcon_project/users/models/custom_user.py
simcon_project/users/models/student.py
simcon_project/users/models/subject_label.py
simcon_project/users/static/users/student_home.css
simcon_project/users/tables/__init__.py
simcon_project/users/tables/student_home_table.py
simcon_project/users/views/redirect_from_login.py
simcon_project/users/views/settings.py
simcon_project/users/views/student_home.py
simcon_project/users/views/template_start.py

devin-w-hill committed:
Merge remote-tracking branch 'origin/SYMCO-20-student-landing-page' into SYMCO-20-student-landing-page

# Conflicts:
#	simcon_project/users/models/student.py
#	simcon_project/users/models/subject_label.py

devin-w-hill committed:
adding in files that were deleted while fixing PR for SYMCO 10.

devin-w-hill committed:
adding in files that were deleted while fixing PR for SYMCO 10.

JerCarnahan reviewed (dismissed):
I see no issues other than the ones already covered

devin-w-hill committed:
moved static directory to toplevel under simcon_project.

devin-w-hill committed:
switched all files to LF instead of CRLF. Linux compatible, should avoid files changed issue (where files showed as changed but weren't)

devin-w-hill committed:
fixed circular import issue and re-did comments in student_home.css using /*...*/ not #

devin-w-hill committed:
Merge branch 'main' into SYMCO-20-student-landing-page

kaigoos reviewed (approved):
Looks good!

Likhovodov reviewed (approved):
Looks good
